### PR TITLE
Script for auto-update leaderboard inactive players

### PR DESCRIPTION
**Pull request by Brutus5000**
To be scheduled for daily execution

### CODE REVIEW

File: scripts/scheduled-leaderboard-purge.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+echo "Reading db config from $DB_CONFIG_FILE"
+
+# shellcheck source=../config/faf-db/faf-db.env"
+source "$DB_CONFIG_FILE"
+
+# Ladder1v1 leaderboard
+echo "Processing inactive users for ladder1v1 leaderboard"
+
+docker exec -i faf-db mysql -D "${MYSQL_DATABASE}" <<SQL_SCRIPT
+  CREATE TEMPORARY TABLE active_players AS
```
Comment (norraxx) [inline scripts/scheduled-leaderboard-purge.sh:15]: Question is: do we need that temporary table? 🙈
We could join on that query as it is and no need to have temporary table.

File: scripts/scheduled-leaderboard-purge.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      (
+          SELECT DISTINCT gps.playerId
+          FROM game_player_stats gps
+          INNER JOIN game_stats gs on gps.gameId = gs.id
+          WHERE gs.endTime > now() - INTERVAL 1 YEAR
+          AND gs.gameMod = 6
+          AND gs.validity = 0
```
Comment (norraxx) [inline scripts/scheduled-leaderboard-purge.sh:22]: what does it mean `gs.validity`?

File: scripts/scheduled-leaderboard-purge.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      );
+
+  UPDATE global_rating
+      LEFT JOIN active_players ON global_rating.id = active_players.playerId
+  SET is_active = active_players.playerId IS NOT NULL
+  WHERE is_active != active_players.playerId IS NOT NULL;
+
+  DROP TABLE active_players;
+SQL_SCRIPT
```
Comment (norraxx) [inline scripts/scheduled-leaderboard-purge.sh]: why do we have 2 execs on `$MYSQL_DATABASE` and third on `faf` as it is?

**Review by norraxx — CHANGES_REQUESTED** [review]

File: scripts/scheduled-leaderboard-purge.sh (hunk 2 above)
Comment (Brutus5000) [reply scripts/scheduled-leaderboard-purge.sh:22]: validity = 0 means the game was valid and rated.
validity > 0 gives a reason why the game was not rated (e.g. AI or uneven teams)

File: scripts/scheduled-leaderboard-purge.sh (hunk 3 above)
Comment (Brutus5000) [reply scripts/scheduled-leaderboard-purge.sh]: because copy&paste error :D fixed

File: scripts/scheduled-leaderboard-purge.sh (hunk 1 above)
Comment (Brutus5000) [reply scripts/scheduled-leaderboard-purge.sh:15]: Probably no, but my experience with MySQL is: if you process large datasets and only need a small result, put it into a temporary table. Also makes analysing easier.

**Pull request merged**